extensions/resteasy-jackson/runtime/src/main/java/io/quarkus/resteasy/jackson/runtime/QuarkusObjectMapperContextResolver.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.resteasy.jackson.runtime;
+
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.ext.ContextResolver;
+import javax.ws.rs.ext.Provider;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.quarkus.arc.Arc;
+
+@Provider
+@Priority(Priorities.USER + 10) // give it a priority that ensures that user supplied ContextResolver classes override this one
+public class QuarkusObjectMapperContextResolver implements ContextResolver<ObjectMapper> {
+
+    private ObjectMapper objectMapper;
+
+    @Override
+    public ObjectMapper getContext(Class<?> type) {
+        if (objectMapper == null) {
```

Comment:
I think this is not thread-safe, a begning race condition can occurs here.
Maybe getting the bean from Arc at postConstruct will be better ?

Reply:
I don't think it really matters to be honest. The returned object will always be the same, so even if multiple threads access it, at works they will both go to Arc and return the same object.

Reply:
If I remember well there is a slight chance of double-null reads/writes that may ends up on some NullPointerException (like one times in a million ...).
Long story: https://shipilev.net/blog/2014/safe-public-construction/ :)

Reply:
Cool link, thanks!